### PR TITLE
Major UI overhaul

### PR DESCRIPTION
**Pull request by MaximParker**
Added UI for the Navbar & menu, Login-page, Messages, and Chat pages.

### CODE REVIEW

File: src/App.svelte [hunk 1]
```diff
@@ -42,14 +44,19 @@
...
+    </PrivateRoute>
 
-  <PrivateRoute path="messages" let:location>
-    <Messages />
-  </PrivateRoute>
+    <PrivateRoute path="messages" let:location>
+      <Messages />
+    </PrivateRoute>
 
+    <PrivateRoute path="developer" let:location>
```
Comment (dweverson) [inline src/App.svelte:55]: Is this a tool for when your building? If so probably don't want to commit it.

File: src/components/DevMenu.svelte [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<script>
+  import { createChatroom, db } from "../utils/api";
```
Comment (dweverson) [inline src/components/DevMenu.svelte:2]: as previous comment from the routing path

File: src/components/Chat.svelte [hunk 3]
```diff
@@ -24,55 +24,42 @@
...
-        db,
-        `messages/${currentChat}/conversations/${signedIn.uid}/messages`
-      ),
-      newMessage
+  let currentConversation = [];
+  $: if (conversationArray) {
+    currentConversation = conversationArray.filter(
+      (entry) => entry.recipient == currentRecipient
     );
-    console.log("Sending message: ", recipientCopy.id, "(recipient's copy)");
```
Comment (dweverson) [inline src/components/Chat.svelte:52]: console.log

File: src/utils/api.js [hunk 4]
```diff
@@ -213,47 +217,40 @@ export const createChatroom = (uid_a, uid_b, displayName_a, displayname_b) => {
...
-  });
+export const sendMessage = async (senderID, recipientID, messageObject) => {
+  const senderCopy = await addDoc(
+    collection(
+      db,
+      `messages/${senderID}/conversations/${recipientID}/messages`
+    ),
+    messageObject
+  );
+  console.log("Sending message: ", senderCopy.id, "(sender's copy)");
```
Comment (dweverson) [inline src/utils/api.js:247]: console log

Comment (dweverson) [inline src/utils/api.js:247]: console log

File: src/utils/api.js [hunk 5]
```diff
@@ -213,47 +217,40 @@ export const createChatroom = (uid_a, uid_b, displayName_a, displayname_b) => {
...
+  const recipientCopy = await addDoc(
+    collection(
+      db,
+      `messages/${recipientID}/conversations/${senderID}/messages`
+    ),
+    messageObject
+  );
+  console.log("Sending message: ", recipientCopy.id, "(recipient's copy)");
```
Comment (dweverson) [inline src/utils/api.js:255]: log

File: src/utils/auth.js [hunk 6]
```diff
@@ -41,16 +41,20 @@ export const updateUserDisplayName = (newUsername) => {
     });
 };
 
-export const registerUserByEmail = (email, password, username) => {
+export const registerUserByEmail = (email, password, displayName) => {
   return createUserWithEmailAndPassword(auth, email, password)
     .then((userCredential) => {
-      console.log("REGISTERED NEW USER:\n",userCredential.user);
+      console.log("REGISTERED NEW USER:\n", userCredential.user);
```
Comment (dweverson) [inline src/utils/auth.js:47]: log

File: src/utils/auth.js [hunk 7]
```diff
@@ -41,16 +41,20 @@ export const updateUserDisplayName = (newUsername) => {
...
-        displayName: username,
-      });
-      updateUserDisplayName(username);
+        displayName,
+      })
+      return displayName;
+    })
+    .then((displayName) => {
+      updateUserDisplayName(displayName);
+      // addDisplayNameToDB(displayName);
```
Comment (dweverson) [inline src/utils/auth.js:57]: commented out

**Review by dweverson — CHANGES_REQUESTED** [review]
looks good to me, few comments to address/tidy but nothing of concern. Look forward to seeing the overhaul, you'v put a lot of work in.

File: src/App.svelte (hunk 1 above)
Comment (MaximParker) [reply src/App.svelte:55]: This is a page for developer tools for use while setting things up for demonstration purposes; so any functions that we want to run quickly and easily in button form, we can place there for easy access out of the view

File: src/components/DevMenu.svelte (hunk 2 above)
Comment (MaximParker) [reply src/components/DevMenu.svelte:2]: See above

**Review by dweverson — APPROVED** [review]

**Pull request merged**